app/services/fact_importer.rb

```diff
@@ -18,7 +18,7 @@ def self.excluded_facts_regex
     Setting.convert_array_to_regexp(
       Setting[:excluded_facts],
       {
-        :prefix => '(\A|.*::|.*_)',
+        :prefix => '(\A|.*::)',
```

Comment:
In my opinion, this patch should drop everything except this line.

Then add a test to exercise the problem that the user describes in the issue/bz:

docker*
will also exclude facts like:
blah_docker_blah

Also fix tests, there will be some artificial assertions that are simply not real-world cases.